core/services/sleeper_task.go

```diff
@@ -13,9 +17,10 @@ type Worker interface {
 }
 
 type sleeperTask struct {
-	worker Worker
-	waker  chan struct{}
-	closer chan struct{}
+	worker   Worker
+	waker    chan struct{}
+	closer   chan struct{}
+	workerWg sync.WaitGroup
```

Comment:
should be workerWG -> https://github.com/golang/go/wiki/CodeReviewComments#initialisms